table/server/master/src/main/java/alluxio/master/table/AlluxioCatalog.java

```diff
@@ -435,11 +434,56 @@ public void remove() {
     };
   }
 
+  private Iterator<Journal.JournalEntry> getDbInfoIterator() {
+    final Iterator<Map.Entry<String, Database>> it = mDBs.entrySet().iterator();
+    return new Iterator<Journal.JournalEntry>() {
+      private Map.Entry<String, Database> mEntry = null;
+
+      @Override
+      public boolean hasNext() {
+        if (mEntry != null) {
+          return true;
+        }
+        if (it.hasNext()) {
+          mEntry = it.next();
+          return true;
+        }
+        return false;
+      }
+
+      @Override
+      public Journal.JournalEntry next() {
+        if (!hasNext()) {
+          throw new NoSuchElementException();
+        }
+        Database database = mEntry.getValue();
+        mEntry = null;
+        DatabaseInfo info = database.getDatabaseInfo();
+        return Journal.JournalEntry.newBuilder().setUpdateDatabaseInfo(
+            alluxio.proto.journal.Table.UpdateDatabaseInfoEntry.newBuilder()
+                .setDbName(database.getName())
+                .setOwnerName(info.getOwnerName())
+                .setOwnerType(info.getOwnerType())
+                .setComment(info.getComment())
+                .setLocation(info.getLocation())
+                .putAllParameter(info.getParameters()).build()).build();
+      }
+
+      @Override
+      public void remove() {
+        throw new UnsupportedOperationException(
+            "GetDbInfoIteratorr#Iterator#remove is not supported.");
+      }
+    };
+  }
+
   @Override
   public Iterator<Journal.JournalEntry> getJournalEntryIterator() {
     List<Iterator<Journal.JournalEntry>> componentIters = StreamUtils
         .map(JournalEntryIterable::getJournalEntryIterator, mDBs.values());
```

Comment:
why can't the `mDBs.values()` also add the `DBInfo` entries?

Reply:
Here we are calling `getJournalEntryIterator` on every db, `mDBs.values` is a list of all DB objects.

Reply:
Yes, so when we are returning journal entries for a database (the table entries), can we also return the dbInfo one as well?